tools/Utilities/HostsUtility/ViewModels/HostsFileEditorSettingsViewModel.cs

```diff
@@ -0,0 +1,124 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Runtime.CompilerServices;
+using CommunityToolkit.Mvvm.ComponentModel;
+using DevHome.Common.Contracts;
+
+namespace DevHome.HostsFileEditor.ViewModels;
+
+public class HostsFileEditorSettingsViewModel : ObservableObject
+{
+    private readonly ILocalSettingsService localSettingsService;
+
+    private bool _showStartupWarning;
+
+    private int _additionalLinesPosition;
+
+    private bool _loopbackDuplicates;
+
+    private int _encoding;
+
+    public bool ShowStartupWarning
+    {
+        get
+        {
+            var settingName = "HostsUtilitySettings" + nameof(ShowStartupWarning);
+            if (localSettingsService.HasSettingAsync(settingName).Result)
+            {
+                _showStartupWarning = localSettingsService.ReadSettingAsync<bool>(settingName).Result;
+            }
+            else
+            {
+                _showStartupWarning = true;
+            }
+
+            return _showStartupWarning;
+        }
+
+        set
+        {
+            if (_showStartupWarning != value)
+            {
+                NotifyPropertyChanged(value);
+                _showStartupWarning = value;
+            }
+        }
+    }
+
+    public int AdditionalLinesPosition
+    {
+        get
+        {
+            _additionalLinesPosition = GetPropertyValueFromLocalSettings<int>(nameof(AdditionalLinesPosition));
+            return _additionalLinesPosition;
+        }
+
+        set
+        {
+            if (_additionalLinesPosition != value)
+            {
+                NotifyPropertyChanged(value);

```

Comment:
Feels like these 2 lines should be in the reverse order. Is there a reason why you did it this way (so a caller could see the new value in the property change handler but still be able to query the old value)?